--- a/net/minecraft/server/jsonrpc/IncomingRpcMethod.java
+++ b/net/minecraft/server/jsonrpc/IncomingRpcMethod.java
@@ -105,6 +_,8 @@
             }
         }
 
+        /* @deprecated Neo: Use the RegisterEvent to register instead of this  */
+        @Deprecated
         public IncomingRpcMethod<?, ?> register(Registry<IncomingRpcMethod<?, ?>> methodRegistry, String key) {
             return this.register(methodRegistry, Identifier.withDefaultNamespace(key));
         }
@@ -143,18 +_,24 @@
                         paramsJsonElement = jsonArray.get(0);
                     }
 
+                    com.mojang.serialization.DynamicOps<JsonElement> ops;
+                    try {
+                        ops = minecraftApi.getServer().registryAccess().createSerializationContext(JsonOps.INSTANCE);
+                    } catch (net.minecraft.server.jsonrpc.methods.MethodNotFoundJsonRpcException e) {
+                        ops = JsonOps.INSTANCE;
+                    }
                     Params params = this.info
-                        .params()
-                        .get()
-                        .schema()
-                        .codec()
-                        .parse(JsonOps.INSTANCE, paramsJsonElement)
-                        .getOrThrow(InvalidParameterJsonRpcException::new);
+                            .params()
+                            .get()
+                            .schema()
+                            .codec()
+                            .parse(ops, paramsJsonElement)
+                            .getOrThrow(InvalidParameterJsonRpcException::new);
                     Result result = this.function.apply(minecraftApi, params, clientInfo);
                     if (this.info.result().isEmpty()) {
                         throw new IllegalStateException("No result codec defined");
                     } else {
-                        return this.info.result().get().schema().codec().encodeStart(JsonOps.INSTANCE, result).getOrThrow(EncodeJsonRpcException::new);
+                        return this.info.result().get().schema().codec().encodeStart(ops, result).getOrThrow(EncodeJsonRpcException::new);
                     }
                 }
             } else {
@@ -176,12 +_,18 @@
                     if (this.info.result().isEmpty()) {
                         throw new IllegalStateException("No result codec defined");
                     } else {
+                        com.mojang.serialization.DynamicOps<JsonElement> ops;
+                        try {
+                            ops = minecraftApi.getServer().registryAccess().createSerializationContext(JsonOps.INSTANCE);
+                        } catch (net.minecraft.server.jsonrpc.methods.MethodNotFoundJsonRpcException e) {
+                            ops = JsonOps.INSTANCE;
+                        }
                         return this.info
                             .result()
                             .get()
                             .schema()
                             .codec()
-                            .encodeStart(JsonOps.INSTANCE, result)
+                            .encodeStart(ops, result)
                             .getOrThrow(InvalidParameterJsonRpcException::new);
                     }
                 }
